mpas_analysis/shared/plot/plotting.py

```diff
@@ -27,6 +28,192 @@
 
 from ..constants import constants
 
+
+def nino34_spectra_plot(config, f, ninoSpectra, confidence95, confidence99,
+			redNoiseSpectra, title, fileout, linewidths,
+                        xlabel='Period (years)', ylabel='Power Spectrum',
+                        titleFontSize=None, figsize=(15, 6), dpi=300):
+    """
+    Plots the nino34 time series and power spectra in an image file
+    Parameters
+    ----------
+    config : instance of ConfigParser
+        the configuration, containing a [plot] section with options that
+        control plotting
+
+	f : frequencies to plot on x-axis
+
+    ninoSpectra : nino34 power spectra
+
+    confidence95 : 95% confidence level based on chi squared test
+
+    confidence99 : 99% confidence level based on chi squared test
+
+    redNoiseSpectra : red noise fit to the ninoSpectra
+
+    title : str
+        the title of the plot
+
+    xlabelTimeSeries, ylabelTimeSeries, xlabelSpectra, ylabelSpectra : str
+        axis labels
+
+    fileout : str
+        the file name to be written
+
+    linewidths : control line width
+
+    titleFontSize : int, optional
+        the size of the title font
+
+    figsize : tuple of float, optional
+        the size of the figure in inches
+
+    dpi : int, optional
+        the number of dots per inch of the figure
+
+    Author
+    ------
+    Luke Van Roekel
+
+    Last Modified
+    -------------
+    03/21/2017
+    """
+
+    fig, ax = plt.subplots()
+
+    ax.plot(f[2:-3], ninoSpectra[2:-3], 'k', linewidth=linewidths)
+    ax.plot(f[2:-3], redNoiseSpectra[2:-3], 'r', linewidth=linewidths)
+    ax.plot(f[2:-3], confidence95[2:-3], 'b', linewidth=linewidths)
+    ax.plot(f[2:-3], confidence99[2:-3], 'g', linewidth=linewidths)
+    ax.set_xlim(15, 1)
+
+#   add legend
```

Comment:
I fought against it at first myself.  But now that I'm using an editor that flags formatting as I go, I can really appreciate how much cleaner code can be when it's formatted uniformly as opposed to each file (or each routine) having a different style.  @pwolfram may have more to say, since he was the one who pushed hard for a uniform style.